chain/chain/src/store.rs

```diff
@@ -1150,6 +1164,9 @@ pub struct ChainStoreUpdate<'a> {
     final_head: Option<Tip>,
     largest_target_height: Option<BlockHeight>,
     trie_changes: Vec<WrappedTrieChanges>,
+    // All state changes made by a chunk, this is only used for splitting states
+    add_state_changes_for_split_states: HashMap<(CryptoHash, ShardId), StateChangesForSplitStates>,
```

Comment:
Seems like we only use HashMap here to assert there are no duplicates. Maybe we should use HashSet for removals for the same reason?